rust/src/smb/detect.rs

```diff
@@ -201,3 +201,20 @@ pub extern "C" fn rs_smb_tx_get_dce_iface(state: &mut SMBState,
     }
     return 0;
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_smb_tx_get_filename(tx: &mut SMBTransaction,
+                                                buffer: *mut *const u8,
+                                                buffer_len: *mut u32)
+                                                -> u8
+{
+    if let Some(SMBTransactionTypeData::CREATE(ref x)) = tx.type_data {
```

Comment:
Why only create ?
Are there not commands such as get file info where this may be interesting ?

Reply:
The file name is only transferred in the create request (that allows to create or open files) to get a handle for the target file, other requests like query_info used this handle, so only create is interesting to the get the filename.

Reply:
Oh thanks :-)